### ISSUE TITLE
Crash when creating new project ("Browse" button)

### ISSUE THREAD

**Issue by jkrei0** [godotengine/godot#92401]
### Tested versions

- Crash reproducible since 4.0 (3.2 works, 3.5.3 works, 4.0 crashes, 4.2.2 crashes)
- If I crash 4.x, then try to open 4.x it immediately closes (3.5.3 doesn't open at all, 3.2 works)

### System information

Windows 10 (19045) - Godot v4.2.2 stable - AMD RX6400 GPU, drivers v23.11.1

### Issue description

Godot crashes when creating a new project. If you select "new project", then "browse", it crashes.
If you attempt to open Godot again, it shows a window, then immediately crashes.

Restarting sometimes fixes the issue, clearing %appdata% (local and roaming) almost always fixes the issue.
If you manage to crash Godot while working on a project, it will also not reopen until clearing appdata and restarting.

Note, v3.5.3 does not crash, but won't open after crashing 4.2.2 (it hangs without showing a window). 3.2 will always open.

See video:

https://github.com/godotengine/godot/assets/51007423/4cdd0be7-5e71-44ef-a2f8-6a3f5d446310

### Steps to reproduce

- Open Godot
- Click "New Project"
- Click "browse" to select a path
- Godot crashes

### Minimal reproduction project (MRP)

N/A

**Comment by RealMadvicius** [comment]
I tested all versions since 4.0 till 4.2.2 and could not reproduce this crash.

Maybe you have something like a security app that has some rules set on folders preventing to write or browse, or some of your folders and/or files contained in your folder "My Documents" have a special character or path breaking the browser, I dont remember if godot project browser put things in cache just from browsing though.

Windows 10 has been having very weird behaviour with explorer and browsing or accessing folders depending the way you access them, i had personnaly issues opening new explorer windows from right clicking "documents" open in new window from an existing explorer window, it would crash the whole windows explorer ( i hope it is not one of those nasty bug you have )

**Comment by jkrei0** [comment]
Yeah, I suspect it might be a problem related to my windows user account -- I've had issues with other applications because of poor choices I made (not so nasty to crash explorer, thankfully). I just would've hoped for at least an error message.

I don't have antivirus other than windows defender, so it shouldn't be that. I don't think it's a problem with my Documents folder specifically, since it does that even if I have a different path (on a different drive) typed in before clicking browse.

I'll try to setup a debugger and see if I can get anything out of it.

**Comment by jkrei0** [comment]
I was expecting building/debugging it myself to be painful and it was quite easy. Kudos to the people who did that.

My debug debug build .. works fine and I can't reproduce it to get a stack trace. I'm not sure what that means, but at least I have a working copy? I'll experiment a bit more tomorrow.

**Comment by RealMadvicius** [comment]
@jkrei0 any news about if you managed to reproduce your issue again with a debug build so devs could get more info about the crash you were having?

**Comment by jkrei0** [comment]
Thanks for reminding me to follow up - it works on my laptop, so I've been ignoring the problem.
I did some additional testing, and have some findings:
- `be56cab58` (latest commit when I built it) doesn't crash. Even if I crash an earlier version, it still works fine. So maybe the issue is fixed?
- `15073afe38` (4.2.2.stable) crashes, when downloading from the website, or when building from source. I got it to dump a stack trace, and it seems like it's related to pen/stylus drivers (I have a Huion Kamvas Pro 12 GT-116 if it makes any difference). I've been using a stylus (Wacom Bamboo Ink) on my laptop without issues.
- It takes quite a while to build godot on my computer, and it seems fixed in the latest commit, so I haven't bothered to test it on any other commit. But `c6f1f614` (between the two versions I tested) seems like it may have something to do with it?

```
(gdb) run
Starting program: H:\projects\godot\godot_source\bin\godot.windows.editor.dev.x86_64.console.exe
[New Thread 16236.0x994]
[New Thread 16236.0x29b0]
[New Thread 16236.0x37ac]
Godot Engine v4.2.2.stable.custom_build.15073afe3 - https://godotengine.org

================================================================
CrashHandlerException: Program crashed
Engine version: Godot Engine v4.2.2.stable.custom_build (15073afe3856abd2aa1622492fe50026c7d63dc1)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[0] <couldn't map PC to fn name>
[1] DisplayServerWindows::_update_tablet_ctx (H:\projects\godot\godot_source\platform\windows\display_server_windows.cpp:4232)
[2] DisplayServerWindows::tablet_set_current_driver (H:\projects\godot\godot_source\platform\windows\display_server_windows.cpp:4523)
[3] Main::setup2 (H:\projects\godot\godot_source\main\main.cpp:2485)
[4] Main::setup (H:\projects\godot\godot_source\main\main.cpp:2227)
[5] widechar_main (H:\projects\godot\godot_source\platform\windows\godot_windows.cpp:165)
[6] _main (H:\projects\godot\godot_source\platform\windows\godot_windows.cpp:204)
[7] main (H:\projects\godot\godot_source\platform\windows\godot_windows.cpp:218)
[8] WinMain (H:\projects\godot\godot_source\platform\windows\godot_windows.cpp:232)
[9] __scrt_common_main_seh (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[10] <couldn't map PC to fn name>
-- END OF BACKTRACE --
================================================================
[Thread 16236.0x29b0 exited with code 3221225477]
[Thread 16236.0x37ac exited with code 3221225477]
[Thread 16236.0x994 exited with code 3221225477]

Program terminated with signal SIGSEGV, Segmentation fault.
```

**Comment by RealMadvicius** [comment]
It could be linked to the fact Huion Kamvas Pro 12 GT-116 is a graphic tablet with a display, and possibly it is a driver conflict with how godot expect windows to be handled and rendered and how it communicates with the huion display.
